sql/core/src/main/scala/org/apache/spark/sql/SparkSessionExtensions.scala

```diff
@@ -111,11 +112,12 @@ class SparkSessionExtensions {
   type FunctionDescription = (FunctionIdentifier, ExpressionInfo, FunctionBuilder)
   type TableFunctionDescription = (FunctionIdentifier, ExpressionInfo, TableFunctionBuilder)
   type ColumnarRuleBuilder = SparkSession => ColumnarRule
-  type QueryStagePrepRuleBuilder = SparkSession => Rule[SparkPlan]
+  type QueryStageRuleBuilder = SparkSession => Rule[SparkPlan]
```

Comment:
- Although `QueryStagePrepRuleBuilder` and `QueryStageRuleBuilder` types has identical types for now, do we need to merge like this?
- Why don't we keep the existing `QueryStagePrepRuleBuilder` in the AS-IS status and add `QueryStageRuleBuilder` newly? Then, we can change both types independently in the future.

Reply:
Although `SparkSessionExtensions` is experimental and we can change dramatically, I believe we had better keep the existing `type` and methods as much as possible for the users. WDYT, @ulysses-you ?

Reply:
thank you @dongjoon-hyun for the input, I agree. It would be better to add a new one.